robolectric/src/main/java/org/robolectric/android/internal/AndroidTestEnvironment.java

```diff
@@ -403,6 +404,11 @@ private Package loadAppPackage_measured(Config config, AndroidManifest appManife
       Path packageFile = appManifest.getApkFile();
       parsedPackage = ShadowPackageParser.callParsePackage(packageFile);
     }
+    if (parsedPackage != null
```

Comment:
Is this for legacy resources only? I think for binary resources applicationInfo.appComponentFactory should already be populated?

Reply:
I remember we don't populate `applicationInfo.appComponentFactory` for binary resources. I wrote an article to analyze `AppComponentFactory` last weekend, and we need to parse `appComponentFactory` from `AndroidManifest.xml` and pass it to `ApplicationInfo` manually. And I don't find any existing code that parses `appComponentFactory` and passes it to `ParsedPacakge`'s `ApplicationInfo`. If I miss something please let me know.